test/ably/restchannelpublish_test.py

```diff
@@ -348,3 +348,13 @@ def test_wildcard_client_id_can_publish_as_others(self):
 
         self.assertEqual(messages[0].client_id, some_client_id)
         self.assertIsNone(messages[1].client_id)
+
+    def test_publish_message_with_connection_key(self):
+        publish0 = self.ably.channels['testConnectionKey']
+
+        with self.assertRaises(AblyException) as context:
+            # publishing the message with an invalid key must fail
+            publish0.publish("publish",
```

Comment:
You should be checking the error code and possibly the error message to be sure it's failing because of the `connection_id` specifically.  Otherwise, this test is very brittle and is not necessarily confirming that the connection_id is being used as expected